Sources/Swiftly/Config.swift

```diff
@@ -56,7 +56,12 @@ public struct Config: Codable, Equatable {
             error: \"\(error)\".
             To use swiftly, modify the configuration file to fix the issue or perform a clean installation.
             """
+            #if !os(macOS)
             throw Error(message: msg)
+            #else
+            let pd = PlatformDefinition.init(name: "xcode", nameFull: "osx", namePretty: "macOS", architecture: nil)
+            return Config.init(inUse: nil, installedToolchains: [], platform: pd)
+            #endif
```

Comment:
The motivation was to have Swiftly be capable of being run without the installer script in dev, or even through other types of installers (e.g. homebrew). The test strategy for macOS is manual at the moment, and there's no release for it yet for the install script to download. There needs to be a way to get a swiftly binary onto a Mac and run through tests.

MacOS doesn't require all of the auto-detection of platform like Linux. This all becomes more uniform in the future with Swiftly self-install through the init subcommand.

Reply:
The existing way to do this is to set the following environment variables:
- SWIFTLY_PLATFORM_NAME
- SWIFTLY_PLATFORM_NAME_FULL
- SWIFTLY_PLATFORM_NAME_PRETTY

You can see example values for these files in the various dockerfiles that were previously used for CI: 
https://github.com/swiftlang/swiftly/blob/ab38db009304334aebd3c661126eba8b2b1536b7/docker/docker-compose.2204.yaml#L14-L16 

I think it'd be best if we continued to rely on these environment variables for this, since changing this code might affect non-test usage of swiftly.